### PR TITLE
[AU4] Fix project properties constructor

### PR DESCRIPTION
**Pull request by grliszas14**
Resolves: https://github.com/audacity/audacity/issues/6991

### CODE REVIEW

File: au4/src/projectscene/qml/Audacity/ProjectScene/clipsview/TracksClipsView.qml [hunk 1]
```diff
@@ -63,6 +63,7 @@ Rectangle {
         timeline.init()
         playCursorController.init()
         tracksViewState.init()
+        project.load()
```
Comment (igorkorsukov) [inline au4/src/projectscene/qml/Audacity/ProjectScene/clipsview/TracksClipsView.qml:66]: It's better to do the opposite here, call init, in which to call load.
init need call it once.
Load can be called several times for some reason.

File: au4/src/project/view/projectpropertiesmodel.cpp [hunk 2]
```diff
@@ -36,13 +36,22 @@ ProjectPropertiesModel::ProjectPropertiesModel(QObject* parent)
     //     m_projectMetaInfo = project->metaInfo();
     // }
 
+}
+
+void ProjectPropertiesModel::init()
+{
+    if (!m_project) {
```
Comment (igorkorsukov) [inline au4/src/project/view/projectpropertiesmodel.cpp:43]: `m_project = context()->currentProject();` - it's better to call it in init.
Also we should subscribe to project changes in global context

File: au4/src/projectscene/qml/Audacity/ProjectScene/clipsview/TracksClipsView.qml (hunk 1 above)
Comment (grliszas14) [reply au4/src/projectscene/qml/Audacity/ProjectScene/clipsview/TracksClipsView.qml]: fixed

File: au4/src/project/view/projectpropertiesmodel.cpp (hunk 2 above)
Comment (grliszas14) [reply au4/src/project/view/projectpropertiesmodel.cpp:40]: fixed

File: au4/src/project/view/projectpropertiesmodel.cpp [hunk 3]
```diff
@@ -31,14 +31,29 @@ using namespace au::project;
...
+        return;
+    }
+
     // if (project) {
     //     m_projectMetaInfo = project->metaInfo();
     // }
 
+    globalContext()->currentProjectChanged().onNotify(this, [this]() {
+        m_project = globalContext()->currentProject();
+    });
```
Comment (igorkorsukov) [inline au4/src/project/view/projectpropertiesmodel.cpp:50]: Should be before 
```
if (!m_project) {
        **return**;
}
```

Comment (grliszas14) [reply au4/src/project/view/projectpropertiesmodel.cpp]: fixed

File: au4/src/project/view/projectpropertiesmodel.cpp [hunk 4]
```diff
@@ -31,14 +31,29 @@ using namespace au::project;
...
+
+    if (!m_project) {
+        return;
+    }
+
     // if (project) {
     //     m_projectMetaInfo = project->metaInfo();
     // }
```
Comment (igorkorsukov) [inline au4/src/project/view/projectpropertiesmodel.cpp:50]: I lost the point, where do we get the properties now?
And when changing the project in the global context, we need to get new properties and call the load.

Comment (grliszas14) [reply au4/src/project/view/projectpropertiesmodel.cpp:50]: Right now we're not using the properties yet, see the commented metaInfo() function.
I've used the properties model to retrieve path for temporary thumbnail image in QML, we'll need to update this model later when we're going to have some project preferences UI to change meta info (and the thumbnail image path will change too as we implement this one: https://github.com/audacity/audacity/issues/6996 )

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**